cachi2/core/package_managers/pip.py

```diff
@@ -116,6 +118,31 @@ def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, str]:
     :return: Tuple of strings (name, version)
     :raises PackageRejected: If either name or version could not be resolved
     """
+
+    def reject_package(name, version):
+        missing = [attr for attr, value in zip(["name", "version"], [name, version]) if not value]
+        raise PackageRejected(
+            f"Could not resolve package metadata: {', '.join(missing)}",
+            solution=(
+                "Please specify package metadata in a way that Cachi2 understands (see the docs)."
+            ),
+            docs=PIP_METADATA_DOC,
+        )
+
+    pyproject_toml = PyProjectTOML(package_dir)
+    if pyproject_toml.exists():
+        log.info("Extracting metadata from pyproject.toml")
+        if pyproject_toml.check_dynamic_version():
+            raise UnsupportedFeature(
+                "Parsing dynamic metadata from pyproject.toml is not supported."
+            )
+
+        name = pyproject_toml.get_name()
+        version = pyproject_toml.get_version()
+
+        if not (name and version):
+            reject_package(name, version)
```

Comment:
We shouldn't reject the package if pyproject.toml exists and doesn't have name+version

Some people configure only `[tool]`s in pyproject.toml and have the rest of the metadata in setup.py/setup.cfg/other

IMO we can fall back to setup.py -> setup.cfg if either name or version cannot be resolved from pptoml